aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -235,47 +235,38 @@ private Completable initializeStorageAdapter(Context context) {
         ));
     }
 
-    private void waitForInitialization(@NonNull Action onComplete, @NonNull Consumer<DataStoreException> onError) {
-        Completable.create(emitter -> {
-            categoryInitializationsPending.await();
-            emitter.onComplete();
-        })
-                .timeout(LIFECYCLE_TIMEOUT_MS, TimeUnit.MILLISECONDS)
-                .subscribeOn(Schedulers.io())
-                .subscribe(
-                        onComplete::call,
-                        throwable -> onError.accept(new DataStoreException("Request failed because DataStore is not " +
-                                "initialized.", throwable, "Retry your request."))
-            );
+    private Completable waitForInitialization() {
+        return Completable.fromAction(() -> categoryInitializationsPending.await())
+            .timeout(LIFECYCLE_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+            .subscribeOn(Schedulers.io())
+            .doOnError(error -> LOG.error("DataStore initialization timed out.", error));
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public void start(@NonNull Action onComplete, @NonNull Consumer<DataStoreException> onError) {
-        waitForInitialization(() -> {
-            try {
-                orchestrator.start();
-            } catch (DataStoreException exception) {
-                onError.accept(exception);
-                return;
-            }
-            onComplete.call();
-        }, onError);
+        waitForInitialization()
+            .andThen(orchestrator.start())
+            .subscribeOn(Schedulers.io())
+            .subscribe(
+                onComplete::call,
+                error -> onError.accept(new DataStoreException("Failed to start DataStore.", error, "Retry."))
+            );
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public void stop(@NonNull Action onComplete, @NonNull Consumer<DataStoreException> onError) {
-        waitForInitialization(() -> orchestrator.stop()
-                .subscribeOn(Schedulers.io())
-                .subscribe(
-                    onComplete::call,
-                    error -> onError.accept(new DataStoreException("Failed to stop DataStore.", error,
-                            "Retry your request."))), onError);
+        waitForInitialization()
```

Comment:
Should this one happen on `Schedulers.io()` as above?

Reply:
I'm pretty sure it isn't needed since the `waitForInitialization` method switches to `Schedulers.io()`, but I just added it anyway just to be explicit, in case `waitForInitialization` ever changes.